### ISSUE TITLE
Add long-press hotkey(s)

### ISSUE THREAD

**Issue by bhajneet** [ShabadOS/desktop#352]
A more intuitive way to set a new line as main would be to hold the spacebar key. If it is held for 300ms or more, then set the current line as main.

Other examples:

- Holding ctrl+/ could start a new search (clear input)
- **Holding ctrl+y could automatically go to the second value (basically a quick jump to previous shabad/bani)** this would resolve v1 -> v2 frustrations for some users missing the "back" button. You'd basically be in akv, do one shabad, hold ctrl+y and be back in akv.
- Remove toggle of next shabad from last line and simply make the behavior tap stops at last line and hold on last line will advance to next shabad for all users
- Holding esc could open a menu that lets a user pick which cleared screen to use (1 = Waheguru, 2 = gurus, sahibzade, et al., 0 = custom message (written in settings).
- Holding ctrl+c (which should not be the hotkey for navigator btw), should allow customization of copying, so ctrl+c held + 1 would perhaps copy only gurbani.

**Comment by bhajneet** [comment]
These need animations/way for the user to understand what holding the hotkey does before it activates

In the example of holding the key `down`, `right`, `l`, or `left-click` on the last line of a shabad should start to fill up some progress bar of sorts saying "next shabad" or such which when filled activates the long-press hotkey. In the UI with an unhidden controller it could be a new "line" after the last one which slides up and says "Next shabad" and fills up for whatever amount of time we set for long-press hotkeys (I'm going to suggest half a second for now, but definitely something over 300ms)

**Comment by bhajneet** [comment]
I'd also like to see "holding" of down, right, or `l` in an open controller for navigator scroll down the list of lines without actually activating any of them. The viewport would be scrolling, and whatever line is closest to the middle would be highlighting/activating at the end.

When we have  #237 we can make it so long pressing the scrolling of the viewport will switch to a pseudo-selector / non-activated one which requires enter to actually display.

Long press would also close  #177

**Comment by Harjot1Singh** [comment]
I think #177 is still useful for mobile environments, no?

**Comment by bhajneet** [comment]
True, and and probably the same use case for GUI only users/mouse only users.

Perhaps we can have the next shabad always visible but also activatable by long press on the line before it.

**Comment by Harjot1Singh** [comment]
Another general question/thought I have: long-press hotkeys, or most hotkeys for that matter should be accessible through the GUI, given that they're meant to be shortcuts. So it's worth thinking about how we can replicate these behaviours. Perhaps holding down the history icon is what holding ctrl-y maps to

**Comment by bhajneet** [comment]
Yeah that's a good point re: GUI

I've updated the original table

I also think there should be some way to "see" the hold happening. So some progress bar, or circle circumference that fills up clockwise, etc.

**Comment by bhajneet** [comment]
Depends on #394 and #462 / #154 for further evaluation
